### PR TITLE
Allow each authentication method to have its own expiration date, remove from user entity.

### PR DESCRIPTION
**Pull request by arthurpassos**
```
laptop@arthur:~/work/valid_until_per_auth_method$ cmake-build-debug/programs/clickhouse client -mn
ClickHouse client version 24.9.1.1.
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 24.9.1.

Warnings:
 * Server was built in debug mode. It will work slowly.
 * Server was built with address sanitizer. It will work slowly.
 * Delay accounting is not enabled, OSIOWaitMicroseconds will not be gathered. You can enable it using `echo 1 > /proc/sys/kernel/task_delayacct` or by using sysctl.
 * Table system.session_log is enabled. It's unreliable and may contain garbage. Do not use it for any kind of security monitoring.

arthur :) create user user1 identified by '1' valid until '2010-01-01', by '2' valid until '2024-09-28', by '3', by '4' valid until '2010-01-01';

CREATE USER user1 IDENTIFIED BY '1' VALID UNTIL '2010-01-01' BY '2' VALID UNTIL '2024-09-28' BY '3' BY '4' VALID UNTIL '2010-01-01'

Query id: bcec2f75-8b8a-4486-8462-cf9c7440de4b

Ok.

0 rows in set. Elapsed: 0.006 sec. 

arthur :) show create user user1;

SHOW CREATE USER user1

Query id: 2b82c73d-27f9-42d4-933e-2b86e1d58256

   ┌─CREATE USER user1──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
1. │ CREATE USER user1 IDENTIFIED WITH sha256_password VALID UNTIL '2010-01-01 00:00:00', sha256_password VALID UNTIL '2024-09-28 00:00:00', sha256_password, sha256_password VALID UNTIL '2010-01-01 00:00:00' │
   └────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

1 row in set. Elapsed: 0.004 sec. 

arthur :) exit;
Bye.
laptop@arthur:~/work/valid_until_per_auth_method$ cmake-build-debug/programs/clickhouse client -mn --user user1 --password '1'
ClickHouse client version 24.9.1.1.
Connecting to localhost:9000 as user user1.
Code: 516. DB::Exception: Received from localhost:9000. DB::Exception: user1: Authentication failed: password is incorrect, or there is no user with such name.. (AUTHENTICATION_FAILED)

laptop@arthur:~/work/valid_until_per_auth_method$ cmake-build-debug/programs/clickhouse client -mn --user user1 --password '2'
ClickHouse client version 24.9.1.1.
Connecting to localhost:9000 as user user1.
Connected to ClickHouse server version 24.9.1.

arthur :) exit;
Bye.
laptop@arthur:~/work/valid_until_per_auth_method$ cmake-build-debug/programs/clickhouse client -mn --user user1 --password '3'
ClickHouse client version 24.9.1.1.
Connecting to localhost:9000 as user user1.
Connected to ClickHouse server version 24.9.1.

arthur :) exit;
Bye.
laptop@arthur:~/work/valid_until_per_auth_method$ cmake-build-debug/programs/clickhouse client -mn --user user1 --password '4'
ClickHouse client version 24.9.1.1.
Connecting to localhost:9000 as user user1.
Code: 516. DB::Exception: Received from localhost:9000. DB::Exception: user1: Authentication failed: password is incorrect, or there is no user with such name.. (AUTHENTICATION_FAILED)

laptop@arthur:~/work/valid_until_per_auth_method$ 
```

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow each authentication method to have its own expiration date, remove from user entity.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 2c073ec53eb70d6a10d56c60713b22a6acca10e6 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70090/2c073ec53eb70d6a10d56c60713b22a6acca10e6/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/39786f7352cac2eecf1c3d013a5bf0210276979b/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by arthurpassos** [comment]
I need to fix something related to `ASTAuthenticationData` children.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by arthurpassos** [comment]
@alexey-milovidov friendly reminder

File: src/Parsers/ASTLiteral.cpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (thevar1able) [inline src/Parsers/ASTLiteral.cpp:1]: :raised_eyebrow:

Comment (arthurpassos) [reply src/Parsers/ASTLiteral.cpp]: :open_mouth:

**Pull request merged**